material_3_demo/lib/color_box.dart

```diff
@@ -0,0 +1,93 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+
+class ColorBox extends StatefulWidget {
+  const ColorBox(
+      {super.key,
+      required this.label,
+      required this.tone,
+      required this.color,
+      required this.onColor,
+      required this.height,
+      required this.width,
+      this.displayPaletteInfo = false});
+
+  final String label;
+  final String tone;
+  final Color color, onColor;
+  final double height, width;
+  final bool displayPaletteInfo;
+
+  @override
+  State<ColorBox> createState() => _ColorBoxState();
+}
+
+class _ColorBoxState extends State<ColorBox> {
+  bool hovered = false;
+
+  @override
+  Widget build(BuildContext context) {
+    final fonts = Theme.of(context).textTheme;
+    return MouseRegion(
+      onEnter: (_) {
+        if (mounted) setState(() => hovered = true);
+      },
+      onExit: (_) {
+        if (mounted) setState(() => hovered = false);
+      },
+      child: Container(
+        color: widget.color,
+        height: widget.height,
+        width: widget.width,
+        child: DefaultTextStyle(
+          style: fonts.labelSmall!.copyWith(color: widget.onColor),
+          child: Stack(
+            children: [
+              Positioned(
+                top: 10,
+                left: 10,
+                child: Text(widget.label),
+              ),
+              Positioned(
+                bottom: 10,
+                right: 10,
+                child: Text(widget.displayPaletteInfo ? widget.tone : ''),
+              ),
+              if (hovered)
+                Positioned(
+                  top: 0,
+                  right: 0,
+                  child: IconButton(
+                    padding: EdgeInsets.zero,
+                    color: widget.onColor,
+                    tooltip: 'Copy hex color',
+                    icon: const Icon(Icons.copy, size: 24),
+                    onPressed: () async {
+                      final messenger = ScaffoldMessenger.of(context);
+                      // Copy color as hex to clipboard
+                      var hex = '#';
```

Comment:
Here and everywhere else, we can probably change it to `String` instead of `var` based on the style guide [here](https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-using-var-and-dynamic):)

Reply:
Done